client/client_inventory.go

```diff
@@ -68,8 +87,26 @@ func doSubmitInventory(
 
 	r, err := api.Do(req)
 	if err != nil {
-		log.Error("Failed to submit inventory data: ", err)
-		return r, errors.Wrapf(err, "inventory submit failed")
+		if tries != 0 {
+			for try := 0; try < tries; try++ {
+				sleepInterval, _ := GetExponentialBackoffTime(try, time.Duration(tries)*time.Minute)
```

Comment:
Converting `tries` to `time.Duration`..?

Anyway, this will not work correctly, I believe. `GetExponentialBackoffTime` internally keeps track of the number of retries, and will return error once it is exceeded. That's why I think it should have an extra argument. If it's zero, it uses the existing self-calculated number of retries. If it's non-zero, it uses that exact number of retries. Then you also need to base the `for` loop on the return from `GetExponentialBackoffTime`, not directly on the number of retries.

Giving it an extra argument also allows you to easily extend the flag to work with the existing retries as well, which is the point of introducing a common flag instead of one just for inventory.

If you do change it, a couple of new test cases for this inside `TestExponentialBackoffTimeCalculation` would also be great!

And lastly, I think this change actually warrants *two* changelog entries! The second one can just say that the new `RetryPollCount` flag works for existing retries as well. Rest assured then that you have implemented a lot of value! 😉

Reply:
see below, I really wanted to argue a lot, but fixed all (unless I missed something) instead.